packages/share/ios/Classes/SharePlugin.m

```diff
@@ -16,8 +16,12 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
     if ([@"share" isEqualToString:call.method]) {
       NSDictionary *arguments = [call arguments];
+      NSString *shareText;
+      if ([arguments[@"text"] isKindOfClass:[NSString class]] == YES) {
```

Comment:
I suppose we can simplify the condition to
```objective-c
if ([arguments[@"text"] isKindOfClass:[NSString class]]) { ... }
```
However, since the Dart/Java/ObjC sides of a plugin are co-developed, our plugins don't explicitly validate the arguments received. Instead we assume everything is fine on the receiver side, like so:
```objective-c
NSString *shareText = call.arguments[@"text"];
[self share:shareText withController:...];
```
In particular, the type check and the checks for a null or empty string are already done [on the Dart side](https://github.com/flutter/plugins/blob/master/packages/share/lib/share.dart#L31-L32), so there is little point in repeating that here.

Reply:
@mravn-google I have a habit of being extremely cautious with type-safety. This patch could certainly be written in a more succinct way. Let me know if you prefer it with less lines of code.

Reply:
@johnhaitas I would prefer it written more succinctly to match other plugins in this repo, if you don't mind.

Reply:
@mravn-google done with https://github.com/flutter/plugins/pull/602/commits/bf0af0d72b2a792895941d1db2881cc5f44fa54c